fiona/fio/info.py

```diff
@@ -50,15 +56,31 @@ def env(ctx, key):
 @click.option('--bounds', 'meta_member', flag_value='bounds',
               help="Print the boundary coordinates "
                    "(left, bottom, right, top).")
+@click.option('--name', 'meta_member', flag_value='name',
+              help="Print the datasource's name.")
 @click.pass_context
-def info(ctx, input, indent, meta_member):
+def info(ctx, input, indent, meta_member, layer):
+
+    """
+    Print information about a dataset.
+
+    When working with a multi-layer dataset the first layer is used by default.
+    Use the '--layer' option to select a different layer.
+    """
+
     verbosity = (ctx.obj and ctx.obj['verbosity']) or 2
     logger = logging.getLogger('fio')
     try:
         with fiona.drivers(CPL_DEBUG=verbosity>2):
-            with fiona.open(input) as src:
+            with fiona.open(input, layer=layer) as src:
                 info = src.meta
-                info.update(bounds=src.bounds, count=len(src))
+                info.update(bounds=src.bounds, name=src.name)
```

Comment:
@sgillies Forgot to mention that this PR also adds `Collection.name` to the info report to make it less ambiguous.